### PR TITLE
fix: Enable sorting in admin users table 

### LINKED ISSUE #3487: Admin User: Sorting and Email search missing
From the user admin page the sorting of users according to table headers and the search according to email address is not working at https://eventyay.com/admin/users/all.

Screenshot from 2019-08-28 17-50-08

### PR DESCRIPTION
**Pull request by kushthedude**
fixes #3487 

#### Screenshot :
Screenshot 2019-08-29 at 6 06 29 AM

### CODE REVIEW

File: app/routes/admin/users/list.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ export default class extends Route.extend(EmberTableRouteMixin) {
   async model(params) {
     this.set('params', params);
     let filterOptions = [];
-    const searchField = 'firstName';
+    const searchField = 'email';
```
Comment (shreyanshdwivedi) [inline app/routes/admin/users/list.js]: Why not firstName?

Comment (kushthedude) [reply app/routes/admin/users/list.js]: The issue said to search with `eMail`.

Comment (shreyanshdwivedi) [reply app/routes/admin/users/list.js]: Ah ok. Maybe you can update the PR description and follow the proper format.
Please add screen for reference too.

**Comment by kushthedude** [comment]
@CosmicCoder96 Please Review

**Comment by kushthedude** [comment]
All of them are working fine as all of them are defined in user schema.

File: app/controllers/admin/users/list.js [hunk 2]
```diff
@@ -16,6 +16,8 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
         extraValuePaths : ['firstName', 'deletedAt'],
         cellComponent   : 'ui-table/cell/admin/users/cell-first-name',
         width           : 155,
+        isSortable      : true,
```
Comment (CosmicCoder96) [inline app/controllers/admin/users/list.js]: why would you sort users by id, that's random and will confuse people.

Comment (kushthedude) [reply app/controllers/admin/users/list.js]: Actually I wanted to sort them by name, I didn't notice this , I will fix and make changes regarding `firstName` and `id`

**Comment by kushthedude** [comment]
@CosmicCoder96 @mrsaicharan1 @shreyanshdwivedi I have tested all the actions on local regarding the change I have made of `extraRecords` and `Record`, Also the sorting is also working on the basis of `firstName`, Please Review. Thanks

File: app/templates/components/ui-table/cell/admin/users/cell-first-name.hbs [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
 <div class="ui hidden divider"></div>
 <div class="ui horizontal compact basic buttons">
-  {{#ui-popup tagName='a' click=(action props.actions.moveToUserDetails record) content=(t 'View') class='ui icon button'}}
+  {{#ui-popup tagName='a' click=(action props.actions.moveToUserDetails extraRecords.id) content=(t 'View') class='ui icon button'}}
     <i class="unhide icon"></i>
   {{/ui-popup}}
-  {{#unless record.isSuperAdmin}}
-    {{#if (not record.deletedAt)}}
-      {{#ui-popup content=(t 'Edit') click=(action props.actions.openEditModal record) class='ui icon button'}}
+  {{#unless extraRecords.id.isSuperAdmin}}
```
Comment (shreyanshdwivedi) [inline app/templates/components/ui-table/cell/admin/users/cell-first-name.hbs]: How will this work exactly? I don't understand how `id.isSuperAdmin` works? Am I missing something here?

File: app/templates/components/ui-table/cell/admin/users/cell-first-name.hbs [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
+    {{#if (not extraRecords.id.deletedAt)}}
```
Comment (shreyanshdwivedi) [inline app/templates/components/ui-table/cell/admin/users/cell-first-name.hbs]: shouldn't this be `extraRecords.deletedAt`?

Comment (CosmicCoder96) [inline app/templates/components/ui-table/cell/admin/users/cell-first-name.hbs]: This won't work.

Comment (kushthedude) [reply app/templates/components/ui-table/cell/admin/users/cell-first-name.hbs]: Missed it

File: app/templates/components/ui-table/cell/admin/users/cell-first-name.hbs (hunk 3 above)
Comment (kushthedude) [reply app/templates/components/ui-table/cell/admin/users/cell-first-name.hbs]: Nope, I need to watch `isSuperAdmin` too

**Comment by kushthedude** [comment]
@CosmicCoder96 @mrsaicharan1 @shreyanshdwivedi I have tested all the actions on local regarding the change I have made of extraRecords and Record, Also the sorting is also working on the basis of firstName, Please Review.

**Comment by shreyanshdwivedi** [comment]
@kushthedude please add a gif of this fix maybe. It'll be better to review

**Comment by shreyanshdwivedi** [comment]
@kushthedude please update the PR description and add a gif so that this can be reviewed. Just want to make sure that everything is working fine on your side before pulling changes

**Comment by kushthedude** [comment]
@iamareebjamal Review Please

**Comment by iamareebjamal** [comment]
I have little to no knowledge of frontend

**Comment by kushthedude** [comment]
> I have little to no knowledge of frontend

The PR is already approved by @CosmicCoder96 and since I need 3 reviews to self merge PR, You can bypass this procedure I think😅.

**Pull request merged**